kv/txn_coord_sender.go

```diff
@@ -186,19 +193,23 @@ func (tm *txnMetadata) close(trace *tracer.Trace, txn *proto.Transaction, resolv
 			}
 			call.Reply = &proto.InternalResolveIntentResponse{}
 		}
-		// We don't care about the reply channel; these are best
-		// effort. We simply fire and forget, each in its own goroutine.
+		// These are best effort. We simply fire and forget, each in its own
+		// goroutine.
 		ctx := tracer.ToCtx(context.Background(), trace.Fork())
-		stopper.RunAsyncTask(func() {
-			if log.V(2) {
-				log.Infof("cleaning up intent %q for txn %s", call.Args.Header().Key, txn)
-			}
+		if log.V(2) {
+			log.Infof("cleaning up intent %q for txn %s", call.Args.Header().Key, txn)
+		}
+		wg.Add(1)
+		go func() {
 			sender.Send(ctx, call)
+			wg.Done()
 			if call.Reply.Header().Error != nil {
 				log.Warningf("failed to cleanup %q intent: %s", call.Args.Header().Key, call.Reply.Header().GoError())
 			}
-		})
+		}()
 	}
+	defer trace.Epoch("waiting for intent resolution")()
```

Comment:
well the wording implies that we are beginning to wait. not sure if there are timing implications here